Code/Framework/AzCore/AzCore/std/ranges/zip_view.h

```diff
@@ -0,0 +1,402 @@
+
+namespace AZStd::ranges
+{
+    inline namespace zip_view_impl
+    {
+        template<class... Views>
+        class zip_view;
+    }
+
+    template<class... Views>
+    inline constexpr bool enable_borrowed_range<zip_view<Views...>> = (enable_borrowed_range<Views> && ...);
+
+    // views::zip customization point
+    namespace views
+    {
+        namespace Internal
+        {
+            struct zip_view_fn
+            {
+                template <class... Views>
+                constexpr auto operator()(Views&&... views) const
+                {
+                    if constexpr (sizeof...(Views) == 0)
+                    {
+                        return views::empty<tuple<>>;
+                    }
+                    else
+                    {
+                        return zip_view<views::all_t<decltype((declval<Views>()))>...>(AZStd::forward<Views>(views)...);
+                    }
+                }
+
+            };
+        }
+        inline namespace customization_point_object
+        {
+            constexpr Internal::zip_view_fn zip{};
+        }
+    }
+
+    inline namespace zip_view_impl
+    {
+        namespace ZipViewInternal
+        {
+            template<class... Rs>
+            /*concept*/ constexpr bool zip_is_common = (sizeof...(Rs) == 1 && (common_range<Rs> && ...)) ||
+                (!(bidirectional_range<Rs> && ...) && (common_range<Rs> && ...)) ||
+                ((random_access_range<Rs> && ...) && (sized_range<Rs> && ...));
+
+            template<class... Ts>
+            using tuple_or_pair = tuple<Ts...>;
+
+            template<class F, class Tuple>
+            constexpr auto tuple_transform(F&& f, Tuple&& tuple)
+            {
+                auto TransformToResultTuple = [&](auto&&... elements)
+                {
+                    return tuple_or_pair<invoke_result_t<F&, decltype(elements)>...>(
+                        AZStd::invoke(f, AZStd::forward<decltype(elements)>(elements))...);
+                };
+                return AZStd::apply(AZStd::move(TransformToResultTuple), AZStd::forward<Tuple>(tuple));
+            }
+
+            template<class F, class Tuple>
+            constexpr void tuple_for_each(F&& f, Tuple&& tuple)
+            {
+                auto InvokeTupleElement = [&](auto&&... elements)
+                {
+                    (AZStd::invoke(f, AZStd::forward<decltype(elements)>(elements)), ...);
+                };
+                AZStd::apply(AZStd::move(InvokeTupleElement), AZStd::forward<Tuple>(tuple));
+            }
+
+            template<class F, class Tuple1, class Tuple2, size_t... Indices>
+            constexpr decltype(auto) tuple_zip(F&& f, AZStd::index_sequence<Indices...>,
+                Tuple1&& tuple1, Tuple2&& tuple2)
+            {
+                (AZStd::invoke(AZStd::forward<F>(f), AZStd::get<Indices>(tuple1), AZStd::get<Indices>(tuple2)), ...);
+            }
+        }
+
+        template<class... Views>
+        class zip_view
```

Comment:
As a side note, a whole bunch of this can be used to replace various views in the SceneAPI.